internal/aws/awsutil/conn.go

```diff
@@ -183,14 +183,27 @@ func GetAWSConfigSession(logger *zap.Logger, cn ConnAttr, cfg *AWSSessionSetting
 		Endpoint:               aws.String(cfg.Endpoint),
 		HTTPClient:             http,
 	}
-	defaultCredProviders := defaults.CredProviders(config, defaults.Handlers())
-	overrideCredProviders := override.GetCredentialsChainOverride().GetCredentialsChain()
-	credProviders := make([]credentials.Provider, 0, len(defaultCredProviders)+len(overrideCredProviders))
-	credProviders = append(credProviders, defaultCredProviders...)
-	credProviders = append(credProviders, overrideCredProviders...)
-	config.Credentials = credentials.NewCredentials(&credentials.ChainProvider{
-		Providers: credProviders,
-	})
+	// do not overwrite for sts assume role
+	if cfg.RoleARN == "" {
+		defaultCredProviders := defaults.CredProviders(config, defaults.Handlers())
+		overrideCredProviders := override.GetCredentialsChainOverride().GetCredentialsChain()
+		credProviders := make([]credentials.Provider, 0)
+		if cfg.SharedCredentialsFile != nil {
```

Comment:
Need this for when not using default cred file and profile. Ex container insights.